File: web/src/main/java/com/github/keraton/triptr/web/filter/FilterConfiguration.java
package com.github.keraton.triptr.web.filter;

import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FilterConfiguration {

    @Bean
    public FilterRegistrationBean<CounterFilter> loggingFilter(){
        FilterRegistrationBean<CounterFilter> registrationBean
                = new FilterRegistrationBean<>();

        registrationBean.setFilter(new CounterFilter());
        registrationBean.addUrlPatterns("/*");

        return registrationBean;
    }

}

File: web/src/main/java/com/github/keraton/triptr/web/filter/FilterConfiguration.kt
package com.github.keraton.triptr.web.filter

import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class FilterConfiguration {

    @Bean
    open fun loggingFilter(): FilterRegistrationBean<CounterFilter> {
        val registrationBean = FilterRegistrationBean<CounterFilter>()

        registrationBean.filter = CounterFilter()
        registrationBean.addUrlPatterns("/*")

        return registrationBean
    }

}
